### DISCUSSION TITLE
Combine and add the products in the pos window

### DISCUSSION THREAD

**Discussion by osvaldoclasico** [Blair2004/NexoPOS#611] [Ideas]
Good evening Team, a pleasure to greet you.

I would like you to take this request into account for the next update.

Apparently the products can only be combined in the invoice or receipt, it would be very useful if this same function is in the pos window, it would make it extraordinarily easy when making any sale and it would be more organized.

In this way confusion would be avoided and it would facilitate the view of the list of products.

This is the current view:
https://ibb.co/p2Xw8Zv

It would be great if the result could be seen in this way simply by clicking or scanning the product as many times as required:

https://ibb.co/XFx3yCY

thanks in advance

**Comment by Blair2004** [comment]
Hi,
Thank you for you idea. Before considering this we'll be asking you few questions :

- in case we have 2 same product but want to discount one of these product do you believe it can be made if all product are always merged ?

- NexoPOS works with premium module such as Gastro which provide modifiers features. If all product must be merge how to differentiate same products that has different modifiers ?

- in case of 2 same products but for one of these we want to set a different sale price and a specific quantity how can that the made ?

Thanks

**Reply by osvaldoclasico** [reply]
Case #1: If we have two identical products, why would I discount one product and not the other? It doesn't make sense, you simply generate a discount to your purchase order!

Case # 2: on the same invoice, why would you have to sell the same product at a different price? It wouldn't make sense for me as a customer to buy the same product from you and for one to have (x) price and the other to have (y) price. the most logical would be the following: you have a product, you have (x) quantity and you have (X) price. the ideal is to make a general discovery of the product based on the quantities required. Now, if the product has variations, suppose we sell a mobile phone, in this case an Iphone X. You have several colors, Storage capacity. obviously you will not sell at the same price in this case in the breakdown of my receipt (pos) it should be listed in the way it is currently,

1- Iphone X - Red, 35GB

2- Iphone X Blue 150GB

**Reply by Blair2004** [reply]
1 - It happens to discount one product from 2 similar products. In the case of special offer 1 purchased + 1 free.
2 - Same product might have a different price of these items has a design flaw for example. We won't talk about the variation as it's not supported.

I really believe asking for the quantity before adding to the card remain the solution to this matter as for now, the way it works is useful for the system.

**Comment by osvaldoclasico** [comment]
Now I will present you the following scenarios:

1- I am a grocery buyer, I have a huge list of products, among them I have a product (X).

But by mistake the cashier added the same product (X) below the list.
 
Example:

Product (A)
Product (A)
Product (A)
Product (A)
**_product (X)_**
Product (A)
Product (A)
Product (A)
Product (A)
Product (A)
Product (A)
Product (A)
Product (A)
**_product (X)_**

I would have to read the entire list to find the product and then remove it or change the quantity, it is very uncomfortable and more so when time is the main factor in a grocery store

**Reply by ippezshelby** [reply]
I think this requirement request is okay.
+1 from me

**Reply by Blair2004** [reply]
I'll see how to add an option to enable/disable the merge products on the cart.

**Reply by ippezshelby** [reply]
@Blair2004, making it configurable makes it more flexible

**Reply by Blair2004** [reply]
![feature](https://user-images.githubusercontent.com/5265663/159447155-27c340e7-9970-4164-ac76-56a246e03c09.gif)

Added for v4.7.x

**Reply by osvaldoclasico** [reply]
> ![rasgo](https://user-images.githubusercontent.com/5265663/159447155-27c340e7-9970-4164-ac76-56a246e03c09.gif)
> 
> Agregado para v4.7.x

Excellent, this way you can use it in both modes, congratulations !!!

**Reply by ippezshelby** [reply]
nice theme
